perl/p5-net-apns-persistent/Portfile

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           perl5 1.0
+
+perl5.branches      5.26 5.28
+perl5.setup         Net-APNS-Persistent 0.02 ../by-authors/id/A/AU/AUFFLICK
+platforms           darwin
+maintainers         nomaintainer
+license             {Artistic-1 GPL}
+description         Send Apple APNS notifications over a persistent connection.
+
+long_description    Class to create a persistent connection to Apple's \
+    APNS servers.
+
+checksums           rmd160  0468bcf46c73739766332767ffc93a939c4cf2a2 \
+                    sha256  7984e4eaaa1edcddaa868762efeb36406f5ff5b9f0d2455f8c0b217e5429f397 \
+                    size    9755
+
+depends_run-append  \
+                    port:p5.28-class-accessor \
```

Comment:
All of these dependencies need to use ${perl5.major} instead of hardcoding 5.28.

Reply:
@jmroot Thanks—but what do I do about [mail/dovecot2/Portfile](https://github.com/macports/macports-ports/pull/5482/files#diff-29cc1a03af699652b0531360dbd4aeed)'s `p5` dependencies?
```
    depends_run-append \
                    port:p5.28-net-apns-persistent \
                    port:p5.28-privileges-drop
```

I believe that I must add `PortGroup perl5 1.0` to define `${perl5.major}`, but then this will define a bunch of other build stuff that we don't want, right?

Should these dependencies in `mail/dovecot2/Portfile` simply be:
```
    depends_run-append \
                    port:p5-net-apns-persistent \
                    port:p5-privileges-drop
```

I see inconsistent use when I do a search for `"port:p5"` across the repo.

Reply:
Dovecot should pick a perl version to use, and depend on the module ports corresponding to that version. It also needs to make sure its perl scripts run with that specific perl version, because otherwise the modules aren't guaranteed to be available. You should not depend on p5-whatever because that gives you no control of the perl version used.

Reply:
I'm just trying to choose something that's easiest and less prone to break.

I see that `perl5` current points to `5.26`.

Is there a way of figuring out the perl version that `perl5` is using, and just use that in the `Portfile`?

Reply:
That would be more prone to break, because it would make the build non-reproducible (the result would differ depending on when it was built). Currently perl5 defaults to +perl5_28. That's a reasonable choice. The main thing, though, is that you choose a specific perl port to depend on.

Reply:
OK thanks. That's what I tried to do in the latest force-push. Let's see if it builds all the way…